filebeat/crawler/prospector_log.go

```diff
@@ -20,17 +20,14 @@ type ProspectorLog struct {
 	missingFiles   map[string]os.FileInfo
 	prospectorList map[string]harvester.FileStat
 	config         cfg.ProspectorConfig
-	channel        chan *input.FileEvent
-	registrar      *Registrar
 }
 
-func NewProspectorLog(config cfg.ProspectorConfig, channel chan *input.FileEvent, registrar *Registrar) (*ProspectorLog, error) {
+func NewProspectorLog(p *Prospector) (*ProspectorLog, error) {
 
 	prospectorer := &ProspectorLog{
-		config:    config,
-		channel:   channel,
-		registrar: registrar,
-		lastscan:  time.Now(),
+		Prospector: p,
```

Comment:
I know. I just prefer to have much fewer params. When we move more parts to the Prospector from prospectorer, I think it could be obsolete in the future.